tests/system_tests_tcp_adaptor.py

```diff
@@ -907,5 +914,164 @@ def test_60_q2_holdoff(self):
+                        'id': cls.test_name}),
+            ('listener', {'role': 'normal',
+                          'port': cls.tester.get_port()}),
+        ]
+        config = Qdrouterd.Config(config)
+        cls.router = cls.tester.qdrouterd(cls.test_name, config, wait=True)
+
+        # Start the echo server. This is the server that the tcpConnector
+        # will be connecting to.
+        server_prefix = "ECHO_SERVER ES_%s" % cls.test_name
+        cls.logger = Logger(title="TcpAdaptor",
+                            print_to_console=True,
+                            save_for_dump=False,
+                            ofilename="../setUpClass/TcpAdaptor_echo_server.log")
+        cls.echo_server = TcpEchoServer(prefix=server_prefix,
+                                        port=cls.tcp_server_port,
+                                        logger=cls.logger)
+        # The router and the echo server are running at this point.
+        assert cls.echo_server.is_running
+
+    @SkipIfNeeded(DISABLE_SELECTOR_TESTS, DISABLE_SELECTOR_REASON)
+    def test_01_mgmt(self):
+        """
+        Create and delete TCP connectors and listeners
+        """
+        LISTENER_TYPE = 'org.apache.qpid.dispatch.tcpListener'
+        CONNECTOR_TYPE = 'org.apache.qpid.dispatch.tcpConnector'
+
+        mgmt = self.router.management
+
+        # When starting out, there should be no tcpListeners or tcpConnectors.
+        self.assertEqual(0, len(mgmt.query(type=LISTENER_TYPE).results))
+        self.assertEqual(0, len(mgmt.query(type=CONNECTOR_TYPE).results))
+
+        connector_name = "ServerConnector"
+        listener_name = "ClientListener"
+
+        mgmt.create(type=LISTENER_TYPE,
+                    name=listener_name,
+                    attributes={'address': self.test_name,
+                                'port': self.tcp_listener_port,
+                                'host': '127.0.0.1'})
+        mgmt.create(type=CONNECTOR_TYPE,
+                    name=connector_name,
+                    attributes={'address': self.test_name,
+                                'port': self.tcp_server_port,
+                                'host': '127.0.0.1'})
+
+        # verify the entities have been created and tcp traffic works
+        self.assertEqual(1, len(mgmt.query(type=LISTENER_TYPE).results))
+        self.assertEqual(1, len(mgmt.query(type=CONNECTOR_TYPE).results))
+
+        # Give a second for the tcpListener to start listening.
+        time.sleep(1)
+        # Start the echo client runner
+        client_runner_timeout = 3
+        runner = EchoClientRunner(self.test_name, 1, self.logger,
+                                  None, None, 100, 1,
+                                  timeout=client_runner_timeout,
+                                  port_override=self.tcp_listener_port)
+        result = None
+
+        # Give some time for the client runner to finish up.
+        time.sleep(client_runner_timeout + 1)
```

Comment:
you should be able to do a runner.wait() instead of sleep(). If it returns None then the client shut down correctly.